joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/Kotlin2Cpg.scala

```diff
@@ -186,4 +137,60 @@ class Kotlin2Cpg extends X2CpgFrontend[Config] {
       }
     }
   }
+
+  private def findJarsIn(dirs: Set[String]) = {
+    val jarExtension = ".jar"
+    dirs.foldLeft(Seq[String]())((acc, classpathEntry) => {
+      val f = File(classpathEntry)
+      val files =
+        if (f.isDirectory) f.list.filter(_.extension.getOrElse("") == jarExtension).map(_.toString)
+        else Seq()
+      acc ++ files
+    })
+  }
+
+  private def entriesForSources(files: Iterable[KtFile], relativeTo: String): Iterable[KtFileWithMeta] = {
```

Comment:
`entriesForSources` and `entriesForConfigFiles` look pretty similar. You may be able to extract shared code here.